x/concentrated-liquidity/internal/math/math.go

```diff
@@ -12,9 +14,15 @@ func Liquidity0(amount sdk.Int, sqrtPriceA, sqrtPriceB sdk.Dec) sdk.Dec {
 	if sqrtPriceA.GT(sqrtPriceB) {
 		sqrtPriceA, sqrtPriceB = sqrtPriceB, sqrtPriceA
 	}
-	product := sqrtPriceA.Mul(sqrtPriceB)
-	diff := sqrtPriceB.Sub(sqrtPriceA)
-	return amount.ToDec().Mul(product).Quo(diff)
+
+	// transform sqrtPriceA and sqrtPriceB to BigDec using osmomath for more final precision
+	sqrtPriceAOsmomath := osmomath.MustNewDecFromStr(sqrtPriceA.String())
+	sqrtPriceBOsmomath := osmomath.MustNewDecFromStr(sqrtPriceB.String())
```

Comment:
We should do the conversions from `Dec` to `BigDec`

Reply:
Offline, agreed to revert liquidity0 and liquidity1 back to normal sdk.Decs due to minimal improvement in precision